File: Swift/LeetCode/ListNode.swift
//
//  ListNode.swift
//  LeetCode
//
//  Created by zhangheng on 2019/12/8.
//  Copyright © 2019 zhangheng. All rights reserved.
//

import Cocoa

public class ListNode:CustomStringConvertible {
    public var val: Int
    public var next: ListNode?
    public init(_ val: Int) {
        self.val = val
        self.next = nil
    }
    
    public var description: String {
        var values:[Int] = []
        var node:ListNode? = self
        while node != nil {
            values.append(node!.val)
            node = node?.next
        }
        return values.description
    }
    
    static func Node(_ values:[Int]) -> ListNode? {
        if values.isEmpty {
            return nil
        }
        let node = ListNode(values.first!)
        var temp:ListNode = node
        for value in values[1..<values.count] {
            temp.next = ListNode(value)
            temp = temp.next!
        }
        return node
    }
}

File: Python/ListNode.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

    def __str__(self):
        if self.next == None:
            return str(self.val)
        des = str(self.val)
        node = self.next
        while node:
            des = des + " -> " + str(node.val)
            node = node.next
        return des

    @staticmethod
    def node(vals: []):
        head = None
        node = None
        for val in vals:
            if head == None:
                head = ListNode(val)
                node = head
            else:
                node.next = ListNode(val)
                node = node.next
        return head